datafusion/optimizer/src/push_down_projection.rs

```diff
@@ -94,23 +94,19 @@ fn optimize_plan(
 
             let mut new_expr = Vec::new();
             let mut new_fields = Vec::new();
+            let mut new_required_columns = HashSet::new();
```

Comment:
Since this is shadowing a variable in the outer scope, I wonder if it would make sense to comment why we are starting with an empty set in this case rather than the pushed down columns?

Reply:
👍Yes, I have added comment to explain it.